kong/plugins/datadog/handler.lua

```diff
@@ -53,6 +53,12 @@ local function log(premature, conf, message)
   if premature then
     return
   end
+ 
+  -- unmatched apis are nil
```

Comment:
On thinking through this, I think it's more sane to put this gate in the `:log` phase handler, rather than this local function. Doing so would mean we don't need to needlessly spend the resources to launch a timer event when we know we're just going to bail out.

We can probably put such a gate before the call to `basic_serializer.serialze`, as we don't need that overhead either; we can simply check for the existing for `ngx.ctx.api` as you've done here with `message.api`.